core/mix/appup.ex

```diff
@@ -53,21 +53,15 @@ defmodule AntikytheraCore.Release.Appup do
     {only_v1, only_v2, diff_pairs} = :beam_lib.cmp_dirs(v1_ebin_dir, v2_ebin_dir)
     diff = reject_unchanged_modules(diff_pairs) |> Enum.map(fn {_f1, f2} -> f2 end)
 
-    # Croma automatically defines a module which might be common across multiple gears,
-    # e.g. `Elixir.Croma.TypeGen.Nilable.Antikythera.Url`.
-    # Such module must not be deleted when upgrading one of the gears
-    # because the other gears are still using that module.
-    only_v1_without_auto_generated =
-      Enum.reject(only_v1, fn beam_file ->
-        beam_file
-        |> read_module_name()
-        |> AntikytheraCore.Version.Gear.auto_generated_module?()
-      end)
-
     file_content = {
       v2_charlist,
       [
-        {v1_charlist, generate_instructions(only_v2, diff, only_v1_without_auto_generated)}
+        # Since some modules might be common across multiple gears,
+        # delete_module instructions might accidentally delete modules in use.
+        # So we don't delete modules in the upgrade instruction.
+        # Note that, as a consequence of the above,
+        # we need to deploy without hot code upgrade to delete really unused modules.
+        {v1_charlist, generate_instructions(only_v2, diff, [])}
```

Comment:
For the same reason in https://github.com/access-company/antikythera/pull/199#discussion_r827684985,
```elixir
{v1_charlist, generate_instructions(only_v1, diff, only_v2)}
```
in the next block should be 
```elixir
{v1_charlist, generate_instructions(only_v1, diff, [])}
```

If both of `gear1` and `gear2` has `module M` and `gear1` depends on `gear2`,  IIUC, the app file for `gear1` doesn't have `module M`. Then, if `gear1` removes the dependency for `gear2`, the appup file for the new `gear1` have `:add_module, M` in upgrade, and `:delete_module, M` in downgrade. The downgrade instruction deletes `module M` which both of `gear1` and `gear2` depends on.

Reply:
Thank you for your comment.
Sorry for not considering downgrade case again.
(Maybe the fix in my previous PR was not enough too...)

I eliminated `:delete_module` by redefining `generate_instructions/2`.
https://github.com/access-company/antikythera/pull/200/commits/b7059e1d9b21b07341c3d4c6fc70311bb958c3d2